src/status_im/contexts/wallet/events.cljs

```diff
@@ -547,3 +551,36 @@
  :wallet/process-watch-only-account-from-backup
  (fn [_ [{:keys [backedUpWatchOnlyAccount]}]]
    {:fx [[:dispatch [:wallet/process-account-from-signal backedUpWatchOnlyAccount]]]}))
+
+(defn process-keypairs
+  [{:keys [db]} [keypairs]]
+  (let [existing-keypairs-by-id               (get-in db [:wallet :keypairs])
+        existing-accounts-by-address          (get-in db [:wallet :accounts])
+        {:keys [removed-keypairs-ids
+                removed-account-addresses
+                updated-keypairs-by-id
+                updated-accounts-by-address]} (data-store/process-keypairs keypairs)
+        updated-keypair-ids                   (set (keys updated-keypairs-by-id))
+        updated-account-addresses             (set (keys updated-accounts-by-address))
+        existing-account-addresses            (set (keys existing-accounts-by-address))
+        new-account-addresses                 (clojure.set/difference updated-account-addresses
+                                                                      existing-account-addresses)
+        old-account-addresses                 (->> (vals existing-accounts-by-address)
+                                                   (filter (fn [{:keys [address key-uid]}]
+                                                             (and (contains? updated-keypair-ids key-uid)
+                                                                  (nil? (get updated-accounts-by-address
+                                                                             address)))))
+                                                   (map :address))]
+    (cond-> {:db (-> db
+                     (assoc-in [:wallet :keypairs]
+                               (-> (partial dissoc existing-keypairs-by-id)
+                                   (apply removed-keypairs-ids)
+                                   (merge updated-keypairs-by-id)))
+                     (assoc-in [:wallet :accounts]
+                               (-> (partial dissoc existing-accounts-by-address)
+                                   (apply (into removed-account-addresses old-account-addresses))
+                                   (merge updated-accounts-by-address))))}
+      (seq new-account-addresses)
+      (assoc :fx refresh-accounts-fx-dispatches))))
```

Comment:
Yeah good point about avoiding sending too many events / requests for fetching the accounts. I initially added those events because the wallet-home screen was showing a blank card for a newly created account, but I didn't debug which event was necessary for that to be shown correctly.

I'll try debugging a bit more and also try using the event logic associated with the `:wallet/process-account-from-signal`. If everything seems to be working I'm happy to refactor that into a new event too 👍